src/PerpetualPositionRouter.sol

```diff
@@ -42,21 +42,34 @@ contract PerpetualPositionRouter {
   // TODO: Should we deposit eth into the Perpetual vault?
   receive() external payable {}
 
-  /// @dev Returns the `sqrtPriceLimitX96` from a `uint168`.
-  /// @param args A `uint168` that contains both the `funcId` and the `sqrtPriceLimitX96` needed to
-  /// open or close a position.
-  function _extractSqrtPriceLimitX96(uint168 args) internal pure returns (uint160) {
-    uint168 mask = (1 << 160) - 1;
-    return uint160(args & mask);
+  /// @dev Returns the `sqrtPriceLimitX96` from a `uint200`.
+  /// @param args A `uint200` that contains the `funcId`, `sqrtPriceLimitX96` and `deadline` needed
+  /// to open or close a position.
+  function _extractSqrtPriceLimitX96(uint200 args) internal pure returns (uint160) {
+    // Remove the deadline paramter and then only keep the positive bits in the next 160 bits
+    return uint160((args >> 32) & ((1 << 160) - 1));
+  }
+
+  /// @dev Returns the `deadline` from a `uint200`.
+  /// @param args A `uint200` that contains the `funcId`, `sqrtPriceLimitX96` and `deadline` needed
+  /// to open or close a position.
+  function _extractDeadline(uint200 args) internal pure returns (uint32) {
```

Comment:
Why not merge this method with `_extractSqrtPriceLimitX96` to a single `_extractParams` method that returns a tuple with both values? RR

Reply:
The main reason was separation of concerns, but it doesn't matter anymore as these functions will be removed in the uint optimization pr. Since we will use encodePacked we will no longer need these bitwise operations.